packages/atlas/src/hooks/useBlockTimeEstimation.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+
+import { useJoystream } from '@/providers/joystream'
+
+const ESTIMATED_BLOCK_TIME = 6000
+
+export const useBlockTimeEstimation = () => {
+  const { joystream, proxyCallback } = useJoystream()
+
+  const [currentBlock, setCurrentBlock] = useState(0)
+  const [timeofTheLastBlock, setTimeOfTheLastBlock] = useState(0)
+
+  useEffect(() => {
+    if (!joystream) {
+      return
+    }
+
+    let unsubscribe
+    const init = async () => {
+      unsubscribe = await joystream.subscribeCurrentBlock(
+        proxyCallback((number) => {
+          setCurrentBlock(number)
+          setTimeOfTheLastBlock(Date.now())
+        })
+      )
+    }
+    init()
+
+    return unsubscribe
+  }, [joystream, proxyCallback])
+
+  const convertBlockToDate = (block: number) => {
```

Comment:
I'm wondering, could we make those functions constant? Currently those functions will get updated on every block. If we keep the block number and date inside some ref instead of state, we could make both those convert functions static, so they wouldn't trigger re-renders

Reply:
hmm, I moved basically everything to refs. Let me know what you think.

Reply:
I would move function to regular `useCallback`, other than that, looks good